rllib/agents/maml/maml.py

```diff
@@ -79,30 +78,26 @@ def __init__(self, workers, inner_adaptation_steps, metric_gen):
         self.metrics = {}
         self.metric_gen = metric_gen
 
-    def __call__(self, samples: List[SampleBatchType]):
-        samples, split_lst = self.post_process_samples(samples)
-        self.buffer.extend(samples)
-        self.split.append(split_lst)
-        self.post_process_metrics()
-        if len(self.split) > self.n:
-            out = SampleBatch.concat_samples(self.buffer)
-            out["split"] = np.array(self.split)
-            self.buffer = []
-            self.split = []
-
-            # Metrics Reporting
-            metrics = _get_shared_metrics()
-            metrics.counters[STEPS_SAMPLED_COUNTER] += out.count
-
-            # Reporting Adaptation Rew Diff
-            ep_rew_pre = self.metrics["episode_reward_mean"]
-            ep_rew_post = self.metrics["episode_reward_mean_adapt_" +
-                                       str(self.n)]
-            self.metrics["adaptation_delta"] = ep_rew_post - ep_rew_pre
-            return [(out, self.metrics)]
-        else:
-            self.inner_adaptation_step(samples)
-            return []
+    def __call__(self, itr):
```

Comment:
Can you make this whole thing a plain function instead of a class now?

Reply:
There is still some internal states that needs to be preserved; for example, the number of inner adaptation steps for MAML.